decidim-core/lib/decidim/avatar_generator.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+module Decidim
+  # Generates an avatar on-the-fly for the given user ID. It will always generate
+  # the same avatar for a given user (ie there's no randomness, so it will be
+  # consistent if the avatar is retrieved multiple times. It's intended to be used
+  # together with Dragonfly and avatar_magick.
+  #
+  # Examples:
+  #   AvatarGenerator.new(app, {id: 1, size: "150", background: "cccccc"})
+  #
+  # Note that when one of the optional parameters are invalid (ie negative size,
+  # or a background that is not a valid HEX color), avatar_magick handles this.
+  #
+  class AvatarGenerator
+    # Selected colors form the Google Material Design palette
+    AVATAR_COLORS = %w(
+      e53935 b71c1c d81b60 880e4f 8e24aa 4a148c
+      5e35b1 311b92 3949ab 1a237e 1e88e5 0d47a1 039be5 01579b
+      00acc1 006064 00897b 004d40 43a047 1b5e20 7cb342 33691e
+      c0ca33 827717 fdd835 f57f17 ffb300 ff6f00 fb8c00 e65100
+      f4511e bf360c 6d4c41 3e2723 757575 212121 546e7a 263238
+    ).freeze
+
+    # Initializes the generator.
+
+    # app - a Dragonfly app.
+    # params - A Hash with params from the URL. Needs the `:id` key. Optionals params
+    #   are: `:size`, `:text`, `:background`.
+    def initialize(app, params)
+      @app = app
+      @params = params
+    end
+
+    # Generates the app.
+    def generate
+      app.generate(:initial_avatar, URI.unescape(text), size: size, background_color: background)
+    end
+
+    private
+
+    attr_reader :params, :app
+
+    def user
+      @user ||= Decidim::User.where(id: params[:id]).first
+    end
+
+    def text
+      params[:text] || user&.name || "Anonymous"
```

Comment:
This seems a potential security risk, I'd rather create the avatar in a callback at the model